vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -4566,6 +4567,27 @@ public void scrollToIndex(int rowIndex) {
         getElement().callJsFunction("scrollToIndex", rowIndex);
     }
 
+    /**
+     * Scrolls to the row presenting the given item.
+     * <p>
+     * Note, with lazy loaded content, calling this method may cause performance
+     * issues, as Grid might need to seek through multiple pages and this way
+     * make a number of backend requests. Consider overriding this method with a
+     * solution making a more optimal backend query and then calling the more
+     * lower level scrollToIndex variant.
+     *
+     * @param item
+     *            the item to scroll to
+     */
+    public void scrollToItem(T item) {
```

Comment:
This can be simplified how with the new method in Flow, see https://github.com/vaadin/flow/pull/18306.

Reply:
Based on the eager discussion in this PR, I wonder if it would change anything 🤷‍♂️

Reply:
In normal OSS projects discussions about pull requests happen in the pull request ;-)

As I haven't been involved in the discusssions, I guess somebody else should take this  forward.

BTW. Quickly checking, I'd rather not change this PR at this point, but make another one that then modifies this to take advantage of these "new features". The referred change seems to need a major release and this could be shipped as such in next minor. As a framework user and add-on developer I'd hope next major is not close, but would still love to get new handy features in.

Reply:
The mentioned change is available in `24.4.0.alpha1` and should not require new major release. We can take this forward.